invenio_rdm_records/services/generators.py

```diff
@@ -262,3 +264,14 @@ def __init__(self, config_key, accept_values=[True], **kwargs):
     def _condition(self, **_):
         """Check if the config value is truthy."""
         return current_app.config.get(self.config_key) in self.accept_values
+
+
+class ResourceAccessToken(Generator):
+    """Allow resource access tokens."""
+
+    def needs(self, record=None, file_key=None, **kwargs):
+        """Enabling Needs."""
+        if record and file_key:
+            return [RATNeed("get_content_files", record["id"], file_key)]
```

Comment:
minor: this could also be more generic than `get_content_files`, so something like `files`/`draft_files`, `record`/`draft_record`.